scripts/1-fetch/doaj_fetch.py

```diff
@@ -0,0 +1,591 @@
+HEADER_LANGUAGE = ["TOOL_IDENTIFIER", "LANGUAGE_CODE", "LANGUAGE", "COUNT"]
+HEADER_PUBLISHER = [
+    "TOOL_IDENTIFIER",
+    "PUBLISHER",
+    "COUNTRY_CODE",
+    "COUNTRY_NAME",
+    "COUNT",
+]
+HEADER_SUBJECT_REPORT = [
+    "TOOL_IDENTIFIER",
+    "SUBJECT_CODE",
+    "SUBJECT_LABEL",
+    "COUNT",
+]
+HEADER_YEAR = ["TOOL_IDENTIFIER", "YEAR", "COUNT"]
+
+# CC License types
+CC_LICENSE_TYPES = [
+    "CC BY",
+    "CC BY-NC",
+    "CC BY-SA",
+    "CC BY-ND",
+    "CC BY-NC-SA",
+    "CC BY-NC-ND",
+    "CC0",
+    "UNKNOWN CC legal tool",
+]
+
+# Language code to readable name mapping
+LANGUAGE_NAMES = {
+    "EN": "English",
+    "ES": "Spanish",
+    "PT": "Portuguese",
+    "FR": "French",
+    "DE": "German",
+    "IT": "Italian",
+    "RU": "Russian",
+    "ZH": "Chinese",
+    "JA": "Japanese",
+    "AR": "Arabic",
+    "TR": "Turkish",
+    "NL": "Dutch",
+    "SV": "Swedish",
+    "NO": "Norwegian",
+    "DA": "Danish",
+    "FI": "Finnish",
+    "PL": "Polish",
+    "CS": "Czech",
+    "HU": "Hungarian",
+    "RO": "Romanian",
+    "BG": "Bulgarian",
+    "HR": "Croatian",
+    "SK": "Slovak",
+    "SL": "Slovenian",
+    "ET": "Estonian",
+    "LV": "Latvian",
+    "LT": "Lithuanian",
+    "EL": "Greek",
+    "CA": "Catalan",
+    "IS": "Icelandic",
+    "MK": "Macedonian",
+    "SR": "Serbian",
+    "UK": "Ukrainian",
+    "BE": "Belarusian",
+    "KO": "Korean",
+    "TH": "Thai",
+    "VI": "Vietnamese",
+    "ID": "Indonesian",
+    "MS": "Malay",
+    "HI": "Hindi",
+    "BN": "Bengali",
+    "UR": "Urdu",
+    "FA": "Persian",
+    "HE": "Hebrew",
+    "SW": "Swahili",
+    "AF": "Afrikaans",
+}
+
+
+# Load ISO 3166-1 alpha-2 country codes from YAML file
+def load_country_names():
+    """
+    Load country code to name mapping from YAML file.
+
+    Automatically generates data/iso_country_codes.yaml if missing using
+    dev/generate_country_codes.py. This ensures the script is self-contained
+    and does not require manual file creation by users.
+
+    Returns:
+        dict: Mapping of ISO 3166-1 alpha-2 codes to country names
+    """
+    country_file = shared.path_join(
+        PATHS["repo"], "data", "iso_country_codes.yaml"
+    )
+
+    # Generate country codes file if it doesn't exist
+    if not os.path.isfile(country_file):
+        LOGGER.info("Country codes file not found, generating it...")
+        generate_script = shared.path_join(
+            PATHS["repo"], "dev", "generate_country_codes.py"
+        )
+        try:
+            # Standard library
+            import subprocess
+
+            subprocess.run([sys.executable, generate_script], check=True)
+            LOGGER.info("Successfully generated country codes file")
+        except Exception as e:
+            LOGGER.error(f"Failed to generate country codes file: {e}")
+            raise shared.QuantifyingException(
+                f"Critical error generating country codes: {e}", exit_code=1
+            )
```

Comment:
I don't think you need to log an error here, as the raised exception will log a message to the terminal